common/src/main/java/com/alibaba/nacos/common/task/engine/NacosExecuteTaskExecuteEngine.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.common.task.engine;
+
+import com.alibaba.nacos.api.exception.NacosException;
+import com.alibaba.nacos.common.task.AbstractExecuteTask;
+import com.alibaba.nacos.common.task.NacosTaskProcessor;
+import org.slf4j.Logger;
+
+import java.util.Collection;
+
+/**
+ * Nacos execute task execute engine.
+ *
+ * @author xiweng.yy
+ */
+public class NacosExecuteTaskExecuteEngine extends AbstractNacosTaskExecuteEngine<AbstractExecuteTask> {
+    
+    private final TaskExecuteWorker[] executeWorkers;
+    
+    public NacosExecuteTaskExecuteEngine(String name, Logger logger) {
+        super(logger);
+        int workerCount = findWorkerCount();
+        executeWorkers = new TaskExecuteWorker[workerCount];
+        for (int mod = 0; mod < workerCount; ++mod) {
+            executeWorkers[mod] = new TaskExecuteWorker(name, mod, workerCount);
+        }
+    }
+    
+    private int findWorkerCount() {
```

Comment:
If I remember correctly, there's a special way to do this

Reply:
How to do this?

Reply:
com.alibaba.nacos.common.utils.ThreadUtils#getSuitableThreadCount